axum-macros/src/lib.rs

```diff
@@ -540,6 +540,32 @@ pub fn derive_from_request_parts(item: TokenStream) -> TokenStream {
 /// }
 /// ```
 ///
+/// # Limitations
+///
+/// This macro does not work for associated functions — functions defined in an `impl` block that
+/// don't take `self`:
```

Comment:
```suggestion
/// This macro does not work for associated functions that aren't methods — functions defined in an `impl` block that
/// don't take `self`:
```

Reply:
But an associated function is never a method 🤔

Reply:
I added this clarification @davidpdrsn — maybe I'm splitting hairs, but I think it pays to try and get it right

Reply:
https://doc.rust-lang.org/reference/items/associated-items.html#methods

> Associated functions whose first parameter is named self are called methods and may be invoked using the method call operator, for example, x.foo(), as well as the usual function call notation.

Reply:
```suggestion
/// This macro does not work functions in an `impl` block that
/// don't have a `self` parameter:
```

Reply:
Oh, already merged 😅